velox/type/Filter.h

```diff
@@ -104,6 +110,15 @@ class Filter {
     return 0;
   }
 
+  /**
+   * Only used in test code.
+   * @return Whether an object is the same as itself.
+   */
+  virtual bool testingEquals(const Filter& other) const {
```

Comment:
This method is a bit tricky. It is easy to forget to override it in a concrete Filter implementation and get incorrect results. I wonder if it would be safer to make it pure virtual and provide a different helper method to use in implementation of testingEquals.

Reply:
OK, make it pure virtual and helper method `Filter::testingBaseEquals`